sentry-android-core/src/main/java/io/sentry/android/core/ActivityFramesTracker.java

```diff
@@ -54,6 +54,7 @@ public synchronized void addActivity(final @NotNull Activity activity) {
     if (!isFrameMetricsAggregatorAvailable()) {
       return;
     }
+    frameMetricsAggregator.reset();
```

Comment:
Is it possible we have one activity alive with a transaction happening at the time a new one kicks off? Where the reset breaks reporting the values for the first activity?

Reply:
Actually I think this could be a problem if users disable auto-finish of the activity transactions and finish them manually in onStop. Because when you launch a second Activity, the lifecycle will be like this:

ActivityA->onPause
ActivityB->onCreate
ActivityB->onStart
ActivityB->onResume
ActivityA->onStop

so in this case when the ActivityA transaction gets finished in onStop it will have no metrics (or maybe even the wrong metrics from the ActivityB)

Needs to be tested though, just my assumption based on https://developer.android.com/guide/components/activities/activity-lifecycle#coordinating-activities

Reply:
Hmm need to think some more here. Our other solution was to do the reset after retrieving the counts.

Then this would also cause issues:

```
ActivityA->onPause
ActivityB->onCreate
ActivityB->onStart [frames should start counting here]
ActivityB->onResume
ActivityA->onStop [reset happens; losing counts from ActivityB as well]
```

So counts wouldn't be accurate. Might even be always `0`?

---

Yet another approach was to take snapshots
```
ActivityA->onPause
ActivityB->onCreate [start snapshot for ActivityB happens here]
ActivityB->onStart
ActivityB->onResume
ActivityA->onStop [stop snapshot for ActivityA happens here]
```

When can we call `reset` here and not lose any counts?

Reply:
Maybe this needs to be a multi-step collection process so we get to call reset somewhere?

Reply:
I think that we need to take snapshots and calculate the delta.
That was my suggestion when talking to @romtsn months ago.
startTransaction -> collect data from `frameMetricsAggregator`.
finish Transaction -> collect data from `frameMetricsAggregator`.
Calculate the delta, that's how it should be, I don't see any problems if doing this, or am I missing something?

Reply:
If we're not resetting the data on the `frameMetricsAggregator` the array in there grows. Would need to test if this actually causes problems.

Reply:
After some quick testing:

It stores counts in a `SparseIntArray` thus only adding an entry for each `ms` duration value that any frame took (including a count of many frames fell in that time range). As time goes by this means we'll probably have every possible `ms` length value in the array. For very long transactions (e.g. if a user stayed on a screen for a very long time) and long usage periods on the app this might actually cause problems as we'll have more and more values in the array. So this alone doesn't feel like the right approach to me. Ideally I'd like to get a reset in there somewhere. That's why I suggested we break the tracking apart. Question is if this is worth the effort to invest here if we should rather go with a different appraoch based on `Choreographer` which we could then also use to track UI events etc.

Reply:
yeah feels like custom framecallback approach is the most viable here. We could also persist intermediate counts for ActivityB, so they don't get lost after calling `reset`, and then sum them up with the new counts after the `reset`, but if we `Choreographer` is a silver bullet for them all, I'd go for it probably

Reply:
I agree that calculating the delta with a single `frameMetricsAggregator` should be fine. The distribution of frames should be around 16 ms. I don't expect the `SparseIntArray` to get too big. Even if we have 1000 different frame durations, which is highly unlikely, we only need 1000 * 32 * 2 bits ~ 8 KiB. We never called reset until now, and it seemed not to cause a memory issue.

Reply:
@philipphofmann yeah the case I was worried about was more the one where it tracks until `onActivityStopped` which might me significantly longer.